plugins/inputs/win_wmi/README.md

```diff
@@ -0,0 +1,229 @@
+Server inventory, which includes the patch level and edition of SQL Server that
+is installed.
+
+```toml
+[[inputs.win_wmi]]
+  namespace = "Root\\microsoft\\sqlserver\\ComputerManagement15"
+  classname = "SqlServiceAdvancedProperty"
+  properties = [
+    "PropertyName",
+    "ServiceName",
+    "PropertyStrValue",
+    "SqlServiceType"
+  ]
+  filter = "ServiceName LIKE 'MSSQLSERVER' AND SqlServiceType = 1 AND (PropertyName LIKE 'FILEVERSION' OR PropertyName LIKE 'SKUNAME')"
+  excludenamekey = true
+  name_prefix = "win_wmi_"
+```
+
+## Troubleshooting
+
+If you are getting an error about an invalid WMI namespace, class, or property,
+use the `Get-WmiObject` or `Get-CimInstance` PowerShell commands in order to
+verify their validity. For example:
+
+```powershell
+Get-WmiObject -Namespace root\cimv2 -Class Win32_Volume -Property Capacity, FreeSpace, Name -Filter 'NOT Name LIKE "\\\\?\\%"'
+```
+
+```powershell
+Get-CimInstance -Namespace root\cimv2 -ClassName Win32_Volume -Property Capacity, FreeSpace, Name -Filter 'NOT Name LIKE "\\\\?\\%"'
+```
+
+## Metrics
+
+Fields and tags are dynamically generated based on the structure of the queried
+WMI class. If the WMI class property's value is a string, then the name and
+value will be used as a metric tag. If the WMI class property's value is an
+integer, then the name and value will be used as a metric field.
+
+## Example Output
+
+Some values are changed for anonymity.
+
+```text
+> win_wmi_MBAM_Volume,VolumeName=C:,host=foo Compliant=1i 1654269272000000000
+> win_wmi_MSFT_NetAdapter,Name=Ethernet,host=foo Speed=1000000000i 1654269272000000000
+> win_wmi_SqlServiceAdvancedProperty,PropertyName=FILEVERSION,PropertyStrValue=2019.150.4178.1,ServiceName=MSSQLSERVER,host=foo,sqlinstance=foo SqlServiceType=1i 1654269272000000000
+> win_wmi_SqlServiceAdvancedProperty,PropertyName=SKUNAME,PropertyStrValue=Developer\ Edition\ (64-bit),ServiceName=MSSQLSERVER,host=foo,sqlinstance=foo SqlServiceType=1i 1654269272000000000
+> win_wmi_Win32_ComputerSystem,Domain=company.com,Manufacturer=Lenovo,Model=X1\ Carbon,Name=FOO,host=foo NumberOfLogicalProcessors=20i,NumberOfProcessors=1i,TotalPhysicalMemory=34083926016i 1654269272000000000
+> win_wmi_Win32_OperatingSystem,Caption=Microsoft\ Windows\ 10\ Enterprise,InstallationType=Client,Name=Microsoft\ Windows\ 10\ Enterprise|C:\WINDOWS|\Device\Harddisk0\Partition3,OSArchitecture=64-bit,SiteCode=NYC,host=foo FreeSpaceInPagingFiles=5203244i,FreeVirtualMemory=16194496i,OperatingSystemSKU=4i,ProductType=1i 1654269272000000000
+> win_wmi_Win32_PhysicalMemory,DeviceLocator=DIMM1,Manufacturer=80AD000080AD,Name=Physical\ Memory,PartNumber=HMA82GU6DJR8N-XN\ \ \ \ ,host=foo Capacity=17179869184i,Speed=3200i 1654269272000000000
+> win_wmi_Win32_Processor,Name=Intel(R)\ Core(TM)\ i9-10900\ CPU\ @\ 2.80GHz,host=foo NumberOfCores=10i 1654269272000000000
+> win_wmi_Win32_Volume,Name=C:,host=foo Capacity=511870046208i,FreeSpace=276193509376i 1654269272000000000

```

Comment:
These example metrics have metric name, tag and field names that are a mixture of camel and snake case. Is there any way we could follow the telegraf standard more closely? I understand these are well known names for windows admins, so maybe it's not worth it to try to automatically reformat camel to snake.

Reply:
Metric name is the class name from WMI, so that's Snake_Case, the fields/tags are CamelCase which is also how they come from WMI. I think that's reasonable?